silicon/backends/rabbitmq.hh

```diff
@@ -0,0 +1,260 @@
+#pragma once
+
+# include <stdexcept>
+# include <iostream>
+
+# include <amqp_tcp_socket.h>
+# include <amqp.h>
+# include <amqp_framing.h>
+
+# include <iod/json.hh>
+
+# include <silicon/symbols.hh>
+# include <silicon/service.hh>
+
+iod_define_symbol(hostname)
```

Comment:
Instead of manually defines your symbols use the iod_generate_symbols. It look for the symbol in a list of c++ header and generate the definitions with preprocessing guards to avoid multiple definitions of the same symbol:

Usage: iod_generate_symbols input_cpp_file1, ..., input_cpp_fileN, output_cpp_header_file
You can compile it here: https://github.com/matt-42/iod/tree/master/tools